### ISSUE TITLE
Fetch session restore mode from config dynamically

### ISSUE THREAD

**Issue by Aetf** [shell-pool/shpool#173]
It is now read once and passed to session restore creation in libshpool/src/daemon/shell.rs.

**Comment by ethanpailes** [comment]
I think we may just want to document the current behavior at least while we have the current in-memory term implementation. Currently, users don't have to pay the memory cost of an in-memory terminal if they are using the simple session restore mode, but in order to allow them to dynamically switch to "screen" or "lines" mode we would need to have an in-memory term anyway in that case. Given that shpool_vt100 is a memory hog, I think we should maintain the current behavior until we can replace it with something more polite.

**Comment by Aetf** [comment]
Agreed. We should document what settings can be changed live, what need a daemon restart to take effect.

**Comment by ethanpailes** [comment]
Yeah that's true. Each field should have a comment saying if it can handle live updates.
